docs/ml-collaborative-filtering.md

```diff
@@ -19,6 +19,7 @@ by a small set of latent factors that can be used to predict missing entries.
 algorithm to learn these latent factors. The implementation in `spark.ml` has the
 following parameters:
 
+* *checkpointInterval* helps with recovery when nodes fail and StackOverflow exceptions caused by long lineage. **Will be silently ignored if *SparkContext.CheckpointDir* is not set.** (defaults to 10).
```

Comment:
the wording is a bit severe... do we have to say node failure or stackoverflow (latter should be rare anyway?)

also is this list of param sorted in any way? perhaps add checkpointInterval to the end?

Reply:
Checkpointing exists to better deal with node failure and decrease memory consumption from lineage. This wording is taken from the parameter-comment in the ALS implementation itself, so I think it is fitting.

This list of parameters is both a sub-set and unordered.

Reply:
Nit: StackOverflow exceptions -> either StackOverflowError or stack overflow errors. Also you're nesting `*` and `**` in the markdown; does that work?